Remove redundant if check in testGetAuthCookie

mokagio opened the pull request:
There was already an `XCTAssertEqual` for that value.Fixes #

To test:

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

mokagio committed:
Remove redundant `if` check in `testGetAuthCookie`

There was already an `XCTAssertEqual` for that value.

mokagio committed:
Replace multi-`if` with `XCTAssertEqual`s in `testGetAuthCookie`

This will help us better identify test failures. Rather that the whole
if not running its successful branch, we'll get failures for each
`cookie` property that did not match expectation.

mokagio committed:
HACK CI for faster feedback

mokagio committed:
HACK - Repeat test on failure 10 times

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Rewrite `testGetAuthCookie` async expectation using Nimble

This is to see whether Nimble's approach to waiting for async code
results in less timeouts than XCTest.

mokagio committed:
HACK - Run 20x the amount of unit tests
